tracing-subscriber/src/fmt/format/mod.rs

```diff
@@ -862,6 +894,29 @@ where
             )?;
         }
 
+        let line_number = if self.display_line_number { meta.line() } else { None };
+
+        if self.display_filename {
+            if let Some(filename) = meta.file() {
+                write!(
+                    writer,
+                    "{}{}{}",
+                    dimmed.paint(filename),
+                    dimmed.paint(":"),
+                    if line_number.is_some() { "" } else { " " }
+                )?;
+            }
+        }
+
+        if let Some(line_number) = line_number {
+            write!(
+                writer,
+                "{}{} ",
+                dimmed.paint(line_number.to_string()),
+                dimmed.paint(":")
+            )?;
```

Comment:
I don't think we should be calling `to_string` here; that will allocate a new `String`, which could have a performance impact that it would be nice to avoid. Nothing else in the formatting code currently allocates strings, so I'd really like to avoid this.

I think the `to_string()` call here is necessary to use `Style::paint`, which takes a `Cow<'a, str>` rather than a `fmt::Debug` implementation? We can avoid this by using [`ansi_term`'s `prefix` and `suffix` methods](https://docs.rs/ansi_term/latest/ansi_term/struct.Style.html#method.prefix), changing this code to something like this:
```suggestion
            write!(
                writer,
                "{}{}:{} ",
                dimmed.prefix(),
                line_number,
                dimmed.suffix(),
            )?;
```

If we do that, we'll also need to add `prefix()` and `suffix()` methods to the no-op version of `Style` that's used when ANSI support is disabled, here:
https://github.com/tokio-rs/tracing/blob/989e07f913bf292f071dbab1be4a8c4c69502500/tracing-subscriber/src/fmt/format/mod.rs#L1190-L1198

This would look something like the following:
```rust
#[cfg(not(feature = "ansi"))]
impl Style {
    // ...
    fn prefix(&self) -> &'static str {
        ""
    }

    fn suffix(&self) -> &'static str {
        ""
    }
}
```